tensorboard/plugins/graph/tf_graph/tf-graph-scene.ts

```diff
@@ -257,6 +258,13 @@ class TfGraphScene2
   /** Main method for building the scene */
   _build(renderHierarchy: tf_graph_render.RenderGraphInfo) {
     this.templateIndex = renderHierarchy.hierarchy.getTemplateIndex();
+    renderHierarchy.hierarchy.addListener(
+      HierarchyEvent.TemplatesUpdated,
+      () => {
+        this.templateIndex = renderHierarchy.hierarchy.getTemplateIndex();
+        this._nodeColorsChanged();
+      }
+    );
```

Comment:
(We've talked about this once before; don't know if there was a conclusion we drew from it) I prefer there being explicit destruct/removeListener if possible at all. Did you consider using a removeListener? If you were to add one, it is often easier to remove with a `key` instead of a pointer to a function instance.

Reply:
I do recall us discussing whether to call `removeEventListener()` in the context of a browser event listener.  iirc, the concern was that browsers would not clean up properly and cause extra memory to be retained.  I don't think this is the case, but I wasn't able to prove its veracity at that time.

In this particular PR, the context is slightly different.  `Hierarchy` is a plain JS class, and listener management is done by us (the application), not the browser's renderer.  As long as there is a path from the application's root to the `Hierarchy`, the `Hierarchy` will not be garbage collected, and neither will its listeners.  However, once the `Hierarchy` changes, due to loading a new graph, or expanding a grouped series node, we lose all references to it.  In this PR, the listeners on the Hierarchy are no longer accessible from the application's root, so they should be garbage collected as well.

I think the pattern in this PR boils down to something like https://gist.github.com/psybuzz/f368886b99b542156e5f7280cc9966b7
e.g. clients can add listeners to the hierarchy and never remove them, and once the hierarchy changes, the listeners will not be marked as collectable by the GC